### PR TITLE
[RM] Script to list the contributors between two git revisions

### PR DESCRIPTION
**Pull request by mattam82**
Also mention it in release-process.md for the future.

**Kind:** documentation / infrastructure.

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).

### CODE REVIEW

File: dev/tools/list-contributors.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+# For compat with OSX which has a non-gnu sed which doesn't support -z
+SED=`which gsed || which sed`
+
+if [ $# != 1 ]; then
+  error "usage: $0 rev0..rev1"
+  exit 1
+fi
+
+git shortlog -s -n --group=author --group=trailer:Co-authored-by $1 | cut -f2 | sort -k 2 | grep -v -e "coqbot" -e "^$" > contributors.tmp
```
Comment (gares) [inline dev/tools/list-contributors.sh:10]: I think @coqbot deserve a special mention, instead ;-)

**Review by gares — APPROVED** [review]

File: dev/doc/release-process.md [hunk 2]
```diff
@@ -63,13 +63,9 @@ in time.
...
-  The command that was used in the previous versions to get the list
-  of contributors for this version is `git shortlog -s -n
-  VX.X+alpha..master | cut -f2 | sort -k 2`. Note that the ordering is
-  approximative as it will misplace people with middle names. It is
-  also probably not correctly handling `Co-authored-by` info that we
-  have been using more lately, so should probably be updated to
-  account for this.
+  The `dev/tools/list-contributors.sh` script computes the number and
+  list of contributors between Coq revisions. Typically used with
+  `VX.X+alpha..master` to check the contributors of version `VX.X`.
```
Comment (Zimmi48) [inline dev/doc/release-process.md:68]: Didn't you rather mean the following?
```suggestion
  `VX.X+alpha..vX.X` to check the contributors of version `VX.X`.
```

**Review by Zimmi48 — APPROVED** [review]
So you managed to list both authors and co-authors?! That's wonderful :rocket:

**Comment by Zimmi48** [comment]
What about also adding a line retrieving the reviewers from the `Ack-by` and the `Reviewed-by`? Because the GitHub user names are used, this would require expanding significantly the `.mailmap` file (although @coqbot could be updated to use the real names and e-mail addresses when it is used to merge a PR). This would be a good complement to our current list of maintainers and contributors, recognizing all the people that participate in the reviewing process, even without being maintainers.

**Comment by jfehrle** [comment]
I have to agree with @Zimmi48, we should list reviewers, too, to acknowledge their contribution.

Also would be good for `make-changelog.sh` to prompt for reviewers, e.g. "Names of reviewers who went 'above and beyond' or for extensive reviews:".  Even if they leave it blank, it is a reminder.

**Comment by JasonGross** [comment]
An alternative to expanding mailmap is to fetch emails by username from the GitHub API: https://docs.github.com/en/free-pro-team@latest/rest/reference/users#emails .  Or you could use this to script an update of .mailmap

**Comment by Zimmi48** [comment]
You need to extend the mailmap (though you don't have to commit the update) to have git shortlog return the right results. But yes, we can totally script this update (though it might not be worth to write this script if from now on we switch to using name+email in the `Ack-by` and `Reviewed-by` trailers).

**Comment by Zimmi48** [comment]
Let's not keep this PR open forever. It's already good as is.
@coqbot: merge now

**Pull request merged**